pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -960,45 +960,6 @@ func getHashPolicy(push *model.PushContext, node *model.Proxy, dst *networking.H
 	return consistentHashToHashPolicy(consistentHash)
 }
 
-type envoyRouteType int
-
-const (
-	envoyPath envoyRouteType = iota
-	envoyPrefix
-	envoyRegex
-	envoyCatchAll
-)
-
-// nolint: unparam
-func getEnvoyRouteTypeAndVal(r *route.Route) (envoyRouteType, string) {
```

Comment:
Currently this is used only for cathall logic - so I removed it and combined with the new function I added in other PR as it has tests and if we need to enhance catchall logic in future if it is at one place. But if this function is intended for any other purpose - I can revert this.

Reply:
I dont think it is a good idea to couple match all route with match all conditions.

Reply:
As the usage, i suggest renaming it to `isCatchAllRoute`

Reply:
I can rename this - But just curious, why do you think it is not a good idea to combine both?

Reply:
They are different objects and the implemented logic is different.

Reply:
> They are different objects and the implemented logic is different.

I agree, they are different objects but the implemented logic is same. That was my motivation to combine.

Reply:
But I can separate them, NBD